service-front/module/Application/src/Controller/General/CookiesController.php

```diff
@@ -23,32 +21,7 @@ public function indexAction()
         $cookiePolicy = $this->fetchPolicyCookie($request);
 
         if ($request->isPost()) {
-            $form->setData($request->getPost());
-
-            $cookiePolicyViewed = new SetCookie(self::SUBMITTED_COOKIE_PAGE);
-            $cookiePolicyViewed->setValue('true')
-                ->setHttponly(false)
-                ->setSecure(true)
-                ->setExpires(new \DateTime('+60 seconds'));
-            $this->getResponse()->getHeaders()->addHeaderLine($cookiePolicyViewed->getFieldName(), $cookiePolicyViewed->getFieldValue());
-
-            $cookiePolicy['usage'] = $form->get('usageCookies')->getValue() === 'yes' ? true : false;
-
-            $newCookiePolicy = new SetCookie(self::COOKIE_POLICY_NAME);
-            $newCookiePolicy->setValue(json_encode($cookiePolicy))
-                ->setHttponly(false)
-                ->setSecure(true)
-                ->setPath('/')
-                ->setExpires(new \DateTime('+365 days'));
-            $this->getResponse()->getHeaders()->addHeaderLine($newCookiePolicy->getFieldName(), $newCookiePolicy->getFieldValue());
-
-            $seenCookie = new SetCookie(self::SEEN_COOKIE_NAME);
-            $seenCookie->setValue('true')
-                ->setHttponly(false)
-                ->setSecure(true)
-                ->setPath('/')
-                ->setExpires(new \DateTime('+365 days'));
-            $this->getResponse()->getHeaders()->addHeaderLine($seenCookie->getFieldName(), $seenCookie->getFieldValue());
+            $this->flashMessenger()->addSuccessMessage('You’ve set your cookie preferences.');
```

Comment:
Think this is OK, but it's perhaps misleading: clicking the button has no effect right now, as preferences are saved when clicking the radio buttons, so this flash message is just smoke and mirrors as a consequence. I'd prefer it if clicking the save button made a change to the page with JS, showing the success message, and requiring no trip to the server.

Reply:
Requires a rewrite of the flash box, this will need to be done under a separate ticket.